### PR TITLE
Onboarding phase 1

### LINKED ISSUE #1519: [onboarding] phase 1: TE create view
Phase 1 of onboarding flow.
Phase 2: #1520 
Phase 3: #1521 
Phase 4: #1522 

Add:

- [x] tooltip for start button in log view
- [x] modified TE create view
- [x] modified checkmark button in TE create view
- [x] tooltip for stop button in log view

**Only show this phase to users that are new users and signed up in the app!**

# Steps & designs
## 1. Tooltip in log view pointing user to start button
https://app.zeplin.io/project/5728ce853dd1b182475f25e9/screen/5a8301c783a1a15b4f6e5f6f
screenshot from 2018-02-14 13-41-02

**Show:**
- when user first signs up and sees log view

**Hide:**
- If user has tapped the tooltip to dismiss it
- If user tapped the start button.

Copy: “Tap to start timer”

## 2. New TE view, once user taps “start” for the first time
https://app.zeplin.io/project/5728ce853dd1b182475f25e9/screen/5a84129c0bcd0643489d675a

screenshot from 2018-02-14 13-42-48

Secondary input fields for project and tag adding are removed.
Checkmark button in bottom right is greyed out.

**Show:**
- first time the user starts a TE in the app

## 3. Checkmark button highlighted
https://app.zeplin.io/project/5728ce853dd1b182475f25e9/screen/5a84129b74b47303abb16429
screenshot from 2018-02-14 13-46-28

Checkmark is white, green solid circle around it.

**Show:**
- After step 2, when user starts typing in description field

## 4. Second tooltip in log view pointing to stop button
https://app.zeplin.io/project/5728ce853dd1b182475f25e9/screen/5a8301c63467b6111e837b8d
screenshot from 2018-02-14 13-48-00

**Show:**
- After step 3
- first time user has a running TE in the app

**Hide:**
- If user has tapped the tooltip to dismiss it
- If user tapped the stop button.

### PR DESCRIPTION
**Pull request by simonrozsival**
Ref #1519 

This PR shows a dismissable tooltip right after the user signs up.

Squash whenever

### CODE REVIEW

**Comment by kornelijepetak** [comment]
⚠️ Something's not working here...

Steps I did:
* Signed up with a new email/password
* The #1519.1 tooltip is shown correctly
* I click the play button to start a TE
* A start TE window pops up (#1519.2 is correct: project/tags are hidden)
* I write a description
* I click the green checkmark button (#1519.3 is ok)
* I get returned to the timer screen and **no running TE is visible**
* Note that after sync I can see the TE running in webapp and also I can create other TEs as well
* Syncing after creating some TEs in webapp still doesn't show any TEs on timer screen.

**Comment by simonrozsival** [comment]
🤔

File: Toggl.Daneel/Services/DaneelOnboardingService.cs [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
+﻿using Toggl.Foundation.Services;
 using Toggl.Multivac;
 using Toggl.PrimeRadiant.Settings;
 
 namespace Toggl.Daneel.Services
 {
     public sealed class DaneelOnboardingService : IOnboardingService
     {
-        private readonly IOnboardingStorage storage;
+        public IOnboardingStorage OnboardingStorage { get; }
```
Comment (arturs105) [inline Toggl.Daneel/Services/DaneelOnboardingService.cs]: Why was this made public? I couldn't find any places where it is used.

**Comment by simonrozsival** [comment]
The problem is that https://github.com/toggl/mobileapp/pull/1787/files#diff-5c75665457b5fac5b3662586f015856cR79 never returns the awaited value - I don't understand why, but I will hopefully figure it out soon.

This PR doesn't close the issue - I fixed the description. It should only show the first step - just the "Tap to start timer" tooltip. The others will be in separate PR(s).

**Comment by simonrozsival** [comment]
I will add resetting of the storage once #1791 is merged. I am putting this on hold until this is merged (+ until I find out what is the problem with the `BehaviorSubject` not returning values.

**Comment by simonrozsival** [comment]
@kornelijepetak I was holding it wrong... not it works well.

File: Toggl.PrimeRadiant/Onboarding/IOnboardingStep.cs [hunk 2]
```diff
@@ -1,7 +1,12 @@
-﻿namespace Toggl.PrimeRadiant.Onboarding
+﻿using System;
+using Toggl.PrimeRadiant.Settings;
+
+namespace Toggl.PrimeRadiant.Onboarding
 {
     public interface IOnboardingStep
     {
-        bool ShouldBeVisible { get; }
+        IOnboardingStorage OnboardingStorage { get; }
```
Comment (arturs105) [inline Toggl.PrimeRadiant/Onboarding/IOnboardingStep.cs]: Does this really need to a part of the interface? Isn't this just needed to implement the interface? If so, then it should be made protected and moved to the base onboarding step.

Comment (simonrozsival) [reply Toggl.PrimeRadiant/Onboarding/IOnboardingStep.cs]: I will change it so that it is not needed here.

**Pull request merged**